plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -64,18 +64,52 @@ export const CatalogTable = (props: CatalogTableProps) => {
   const { isStarredEntity, toggleStarredEntity } = useStarredEntities();
   const { loading, error, entities, filters } = useEntityList();
 
-  const defaultColumns: TableColumn<CatalogTableRow>[] = useMemo(
-    () => [
+  const defaultColumns: TableColumn<CatalogTableRow>[] = useMemo(() => {
+    return [
       columnFactories.createNameColumn({ defaultKind: filters.kind?.value }),
-      columnFactories.createSystemColumn(),
-      columnFactories.createOwnerColumn(),
-      columnFactories.createSpecTypeColumn(),
-      columnFactories.createSpecLifecycleColumn(),
+      ...createEntitySpecificColumns(),
       columnFactories.createMetadataDescriptionColumn(),
       columnFactories.createTagsColumn(),
-    ],
-    [filters.kind?.value],
-  );
+    ];
+
+    function createEntitySpecificColumns(): TableColumn<CatalogTableRow>[] {
+      switch (filters.kind?.value) {
+        case 'api':
+          return [
+            columnFactories.createSystemColumn(),
+            columnFactories.createOwnerColumn(),
+            columnFactories.createSpecTypeColumn(),
+            columnFactories.createSpecLifecycleColumn(),
+          ];
+        case 'component':
+          return [
+            columnFactories.createSystemColumn(),
+            columnFactories.createOwnerColumn(),
+            columnFactories.createSpecTypeColumn(),
+            columnFactories.createSpecLifecycleColumn(),
+          ];
+        case 'domain':
+          return [columnFactories.createOwnerColumn()];
+        case 'group':
+          return [columnFactories.createSpecTypeColumn()];
+        case 'location':
+          return [columnFactories.createSpecTypeColumn()];
+        case 'resource':
+          return [
+            columnFactories.createSystemColumn(),
+            columnFactories.createOwnerColumn(),
+            columnFactories.createSpecTypeColumn(),
+            columnFactories.createSpecLifecycleColumn(),
+          ];
+        case 'system':
+          return [columnFactories.createOwnerColumn()];
+        case 'template':
+          return [columnFactories.createSpecTypeColumn()];
+        default:
+          return [];
```

Comment:
shouldn't the default case be same as before? I'm thinking of cases where you have implemented your own type and are happy with all those fields :)